core-tests/src/test/java/org/mule/runtime/core/processor/chain/DefaultMessageProcessorChainTestCase.java

```diff
@@ -87,34 +87,66 @@
 @RunWith(Parameterized.class)
 @SmallTest
 @SuppressWarnings("deprecation")
-public class DefaultMessageProcessorChainTestCase extends AbstractMuleContextTestCase {
+public class DefaultMessageProcessorChainTestCase extends AbstractReactiveProcessorTestCase {
 
   protected MuleContext muleContext;
 
   protected MessageExchangePattern exchangePattern;
-  protected boolean nonBlocking;
   protected boolean synchronous;
   private AtomicInteger nonBlockingProcessorsExecuted = new AtomicInteger(0);
+  private ProcessingStrategyFactory processingStrategyFactory;
+
+  private ExpectedException expectedException = ExpectedException.none();
 
   @Parameterized.Parameters
   public static Collection<Object[]> parameters() {
     return Arrays.asList(new Object[][] {
-        {REQUEST_RESPONSE, false, true},
-        {REQUEST_RESPONSE, false, false},
-        {REQUEST_RESPONSE, true, true},
-        {REQUEST_RESPONSE, true, false},
-        {ONE_WAY, false, true},
-        {ONE_WAY, false, false},
-        {ONE_WAY, true, true},
-        {ONE_WAY, true, false}});
-  }
-
-  private Flow mockFlow = mock(Flow.class);
-
-  public DefaultMessageProcessorChainTestCase(MessageExchangePattern exchangePattern, boolean nonBlocking, boolean synchronous) {
+        {REQUEST_RESPONSE, Mode.BLOCKING, true, new DefaultFlowProcessingStrategyFactory()},
```

Comment:
Mode is protected, don't see why i it's worth making public for static import only.